twake/frontend/src/app/scenes/Client/ChannelsBar/Parts/CurrentUser/SaveNewStatus.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useCallback, useState, useRef } from 'react';
+import { Button } from 'antd';
+
+import Languages from 'services/languages/languages';
+import { useCurrentUser } from 'app/state/recoil/hooks/useCurrentUser';
+import MenusManager from 'app/components/Menus/MenusManager';
+import InputWithIcon from 'app/components/Inputs/InputWithIcon';
+import classNames from 'classnames';
+
+import './SaveNewStatus.scss';
+
+type PropsType = {
+  level: any;
+};
+
+export default (props: PropsType): JSX.Element => {
+  const { user, updateStatus } = useCurrentUser();
+  const icon = user?.status?.split(' ')[0] || '';
+  const text = user?.status?.split(' ').slice(1).join(' ') || '';
+  const [status, setStatus] = useState([icon, text]);
+  const toto = useRef(status);
+
+  const save = useCallback(() => {
+    updateStatus(status);
+    MenusManager.closeMenu();
+  }, [status, updateStatus]);
+
+  useEffect(() => {
+    const listener = (event: { code: string; preventDefault: () => void }) => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        console.log('Enter key was pressed. Run your function.');
+        save();
+      }
+    };
+    document.addEventListener('keydown', listener);
```

Comment:
I don't think you need this useEffect to add an event listener like this as you already use it directly line 48:

``` tsx
 <div
      onKeyDown={evt => {
        if (evt.key === 'Enter') save();
      }}
    >
....
```